api/ruxos_posix_api/src/imp/fd_ops.rs

```diff
@@ -13,30 +13,126 @@ use core::ffi::c_int;
+            st_ctime: rux_timespec::from(cstat.st_ctime),
+            __unused: cstat.__unused,
+        }
+    }
+}
 
-pub trait FileLike: Send + Sync {
-    fn read(&self, buf: &mut [u8]) -> LinuxResult<usize>;
-    fn write(&self, buf: &[u8]) -> LinuxResult<usize>;
-    fn stat(&self) -> LinuxResult<ctypes::stat>;
-    fn into_any(self: Arc<Self>) -> Arc<dyn core::any::Any + Send + Sync>;
-    fn poll(&self) -> LinuxResult<PollState>;
-    fn set_nonblocking(&self, nonblocking: bool) -> LinuxResult;
+impl From<rux_timespec> for ctypes::timespec {
+    fn from(rtimespec: rux_timespec) -> Self {
+        ctypes::timespec {
+            tv_sec: rtimespec.tv_sec,
+            tv_nsec: rtimespec.tv_nsec,
+        }
+    }
+}
+
+impl From<Ruxstat> for ctypes::stat {
+    #[cfg(target_arch = "aarch64")]
+    fn from(rstat: Ruxstat) -> Self {
+        ctypes::stat {
+            st_dev: rstat.st_dev,
+            st_ino: rstat.st_ino,
+            st_mode: rstat.st_mode,
+            st_nlink: rstat.st_nlink,
+            st_uid: rstat.st_uid,
+            st_gid: rstat.st_gid,
+            st_rdev: rstat.st_rdev,
+            __pad: rstat.__pad,
+            st_size: rstat.st_size,
+            st_blksize: rstat.st_blksize,
+            __pad2: rstat.__pad2,
+            st_blocks: rstat.st_blocks,
+            st_atime: rstat.st_atime.into(),
+            st_mtime: rstat.st_mtime.into(),
+            st_ctime: rstat.st_ctime.into(),
+            __unused: rstat.__unused,
+        }
+    }
+
+    #[cfg(any(target_arch = "x86_64", target_arch = "riscv64"))]
+    fn from(rstat: Ruxstat) -> Self {
+        ctypes::stat {
+            st_dev: rstat.st_dev,
+            st_ino: rstat.st_ino,
+            st_nlink: rstat.st_nlink,
+            st_mode: rstat.st_mode,
+            st_uid: rstat.st_uid,
+            st_gid: rstat.st_gid,
+            __pad0: rstat.__pad0,
+            st_rdev: rstat.st_rdev,
+            st_size: rstat.st_size,
+            st_blksize: rstat.st_blksize,
+            st_blocks: rstat.st_blocks,
+            st_atime: rstat.st_atime.into(),
+            st_mtime: rstat.st_mtime.into(),
+            st_ctime: rstat.st_ctime.into(),
+            __unused: rstat.__unused,
+        }
+    }
 }
 
 lazy_static::lazy_static! {
-    static ref FD_TABLE: RwLock<FlattenObjects<Arc<dyn FileLike>, RUX_FILE_LIMIT>> = {
-        let mut fd_table = FlattenObjects::new();
-        fd_table.add_at(0, Arc::new(stdin()) as _).unwrap(); // stdin
-        fd_table.add_at(1, Arc::new(stdout()) as _).unwrap(); // stdout
-        fd_table.add_at(2, Arc::new(stdout()) as _).unwrap(); // stderr
-        RwLock::new(fd_table)
+    static ref must_exec: usize  = {
+        FD_TABLE.write().add_at(0, Arc::new(stdin()) as _).unwrap(); // stdin
+        FD_TABLE.write().add_at(1, Arc::new(stdout()) as _).unwrap(); // stdout
+        FD_TABLE.write().add_at(2, Arc::new(stdout()) as _).unwrap(); // stderr
+        0
```

Comment:
When does this initialization take effect? This should be put into another place (Like `ruxruntime` or somewhere).